example/newreposecret/main.go

```diff
@@ -0,0 +1,179 @@
+//
+// Usage:
+//	export GITHUB_AUTH_TOKEN=<auth token from github that has secret create rights>
+//	export SECRET_VARIABLE=<secret value of the secret variable>
+//	go run main.go -owner <owner name> -repo <repository name> SECRET_VARIABLE
+//
+// Example:
+//	export GITHUB_AUTH_TOKEN=0000000000000000
+//	export SECRET_VARIABLE="my-secret"
+//	go run main.go -owner google -repo go-github SECRET_VARIABLE
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"flag"
+	"fmt"
+	sodium "github.com/GoKillers/libsodium-go/cryptobox"
+	"github.com/google/go-github/v32/github"
+	"golang.org/x/oauth2"
+	"log"
+	"os"
+)
+
+var (
+	repo  = flag.String("repo", "", "The repo that the secret should be added to, ex. go-github")
+	owner = flag.String("owner", "", "The owner of there repo this should be added to, ex. google")
+)
+
+func main() {
+	flag.Parse()
+
+	token := os.Getenv("GITHUB_AUTH_TOKEN")
+	if token == "" {
+		log.Fatal("please provide a GitHub API token via env variable GITHUB_AUTH_TOKEN")
+	}
+
+	if *repo == "" {
+		log.Fatal("please provide required flag --repo to specify GitHub repository ")
+	}
+
+	if *owner == "" {
+		log.Fatal("please provide required flag --owner to speify GitHub user/org owner")
+	}
+
+	secretName, err := getSecretName()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	secretValue, err := getSecretValue(secretName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ctx, client, err := githubAuth(token)
+	if err != nil {
+		log.Fatalf("unable to authorize using env GITHUB_AUTH_TOKEN: %v", err)
+	}
+
+	err = addRepoSecret(ctx, client, *owner, *repo, secretName, secretValue)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("Added secret %q to the repo %v/%v\n", secretName, *owner, *repo)
+}
+
+func getSecretName() (string, error) {
+	secretName := flag.Arg(0)
+	if secretName == "" {
+		err := fmt.Errorf("missing argument secret name")
+		return "", err
+	}
+	return secretName, nil
+}
+
+func getSecretValue(secretName string) (string, error) {
+	secretValue := os.Getenv(secretName)
+	if secretValue == "" {
+		err := fmt.Errorf("secret value not found under env variable %q", secretName)
+		return "", err
+	}
+	return secretValue, nil
+}
+
+// githubAuth returns a GitHub client and context
+// It reads an environment variable GITHUB_AUTH_TOKEN
+// for a GitHub API token with secret read/write permissions
+// cannot be the default token from GitHub actions
+func githubAuth(token string) (context.Context, *github.Client, error) {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+	return ctx, client, nil
+}
+
+// addRepoSecret will add a secret to a GitHub repo for use in GitHub Actions
+//
+// To communicate with GitHub addRepoSecret requires a client and context.
+// To determine what repository to add this secret to it will use the owner and repo combination
```

Comment:
```suggestion
// To determine what repository to add this secret to, it will use the owner and repo combination.
```

This is obvious too, and I think it can be deleted.

Reply:
deleted!